pybamm/models/submodels/interface/open_circuit_potential/plett_ocp.py

```diff
@@ -0,0 +1,151 @@
+            ocp_bulk_eq = self.phase_param.U(sto_bulk, T_bulk)
+
+            c_scale = self.phase_param.c_max
+            variables[f"Total lithium in {phase} phase in {domain} electrode [mol]"] = (
+                sto_bulk * c_scale
+            )  # c_s_vol * L * A
+
+            H = self.phase_param.H(sto_surf)
+            variables[f"{Domain} electrode {phase_name}equilibrium OCP [V]"] = (
+                ocp_surf_eq
+            )
+            variables[f"{Domain} electrode {phase_name}bulk equilibrium OCP [V]"] = (
+                ocp_bulk_eq
+            )
+            variables[f"{Domain} electrode {phase_name}OCP hysteresis [V]"] = H
+
+            dU = self.phase_param.U(sto_surf, T_bulk).diff(sto_surf)
+            # dQ = self.phase_param.Q(sto_surf).diff(sto_surf)
+            dQdU = dQ / dU
+            variables[
+                f"{Domain} electrode {phase_name}differential capacity [A.s.V-1]"
+            ] = dQdU
+            variables[
+                f"{Domain} electrode {phase_name}hysteresis state distribution"
+            ] = h
+            H_x_av = pybamm.x_average(H)
+            h_x_av = pybamm.x_average(h)
+            # check if psd
+            if domain_options["particle size"] == "distribution":
+                # should always be true
+                if f"{domain} particle size" in sto_surf.domains["primary"]:
+                    # check if MPM Model
+                    if "current collector" in sto_surf.domains["secondary"]:
+                        ocp_surf = ocp_surf_eq + H_x_av * h_x_av
+                    # must be DFN with PSD model
+                    elif f"{domain} electrode" in sto_surf.domains["secondary"]:
+                        ocp_surf = ocp_surf_eq + H * h
+                    else:
+                        raise ValueError(
+                            'Model type not implementted with open-circuit potential as "Plett"'
+                        )
+                else:
+                    raise ValueError(
+                        'Model type not implementted with open-circuit potential as "Plett"'
+                    )
+            # must not be a psd
+            else:
+                ocp_surf = ocp_surf_eq + H * h
+            H_s_av = pybamm.size_average(H_x_av)
+            h_s_av = pybamm.size_average(h_x_av)
+
+            ocp_bulk = ocp_bulk_eq + H_s_av * h_s_av
+
+        variables.update(self._get_standard_ocp_variables(ocp_surf, ocp_bulk, dUdT))
+        return variables
+
+    def set_rhs(self, variables):
+        domain, Domain = self.domain_Domain
+        phase_name = self.phase_name
+
+        current = self.param.current_with_time
+        current = variables[f"{Domain} electrode interfacial current density [A.m-2]"]
+        Q_cell = variables[f"{Domain} electrode capacity [A.h]"]
+        dQdU = variables[
+            f"{Domain} electrode {phase_name}differential capacity [A.s.V-1]"
+        ]
+        dQdU = dQdU.orphans[0]
```

Comment:
why do you need to do this?

Reply:
I see. I believe it's because I need the individual phase capacities, so I calculated dQ for each phase individually. dQdU is determined in the coupled variables section, but used in the rhs section. Hopefully the example notebook + the changes in the next commit will be revealing.

Reply:
Succinctly, it is used in the equation below.